src/iproject.ts

```diff
@@ -427,6 +427,88 @@ export class IProject {
     }
   }
 
+  /**
+   * Run the project's build or compile command.
+   * 
+   * @param isBuild True for build command and false for compile command.
+   */
+  public async runBuildOrCompileCommand(isBuild: boolean) {
+    const unresolvedState = await this.getUnresolvedState();
+
+    if (unresolvedState) {
+      const command = isBuild ? unresolvedState.buildCommand : unresolvedState.compileCommand;
```

Comment:
shouldn't we resolve variables in this command string?

Reply:
The `code-for-ibmi.runAction` command provided by Code4i will automatically handle the resolving of variables.